packages/unified-selection/src/test/HiliteSetProvider.test.ts

```diff
@@ -0,0 +1,339 @@
+          yield modelKey;
+          yield categoryKey;
+          yield elementKey;
+        };
+
+        const modelResultKey = createECInstanceId(2);
+        const categoryResultKey = createECInstanceId(2);
+        const elementResultKey = createECInstanceId(2);
+
+        mockClass("TestModel", "BisCore", "Model");
+        mockClass("TestCategory", "BisCore", "Category");
+        mockClass("TestElement", "BisCore", "Element");
+        mockQuery([modelResultKey], [categoryResultKey], [elementResultKey]);
+
+        const selection = Selectables.create([customSelectable]);
+        const result = await loadHiliteSet(selection);
+        expect(result.models).to.deep.eq([modelResultKey]);
+        expect(result.subCategories).to.deep.eq([categoryResultKey]);
+        expect(result.elements).to.deep.eq([elementResultKey]);
+      });
+    });
+
+    describe("Other", () => {
+      it("ignores unknown type", async () => {
+        const elementKey = createSelectableInstanceKey(1);
+        const modelKey = createECInstanceId(2);
+        mockClass("TestClass", "BisCore", "Unknown");
+        mockQuery([modelKey], [modelKey], [modelKey]);
+
+        const selection = Selectables.create([elementKey]);
+        const result = await loadHiliteSet(selection);
+        expect(result.elements).to.be.empty;
+        expect(result.subCategories).to.be.empty;
+        expect(result.models).to.be.empty;
+      });
+
+      it("ignores unknown schema", async () => {
+        metadataProvider.getSchema.returns(Promise.resolve(undefined));
+        const elementKey = createSelectableInstanceKey(1);
+        const resultKey = createECInstanceId(2);
+        mockClass("TestClass", "BisCore", "Element");
+        mockQuery([resultKey], [resultKey], [resultKey]);
+
+        const selection = Selectables.create([elementKey]);
+        const result = await loadHiliteSet(selection);
+        expect(result.elements).to.be.empty;
+        expect(result.subCategories).to.be.empty;
+        expect(result.models).to.be.empty;
+      });
+
+      it("ignores unknown class", async () => {
+        schemaMock.getClass.returns(Promise.resolve(undefined));
+        const elementKey = createSelectableInstanceKey(1);
+        const resultKey = createECInstanceId(2);
+        mockQuery([resultKey], [resultKey], [resultKey]);
+
+        const selection = Selectables.create([elementKey]);
+        const result = await loadHiliteSet(selection);
+        expect(result.elements).to.be.empty;
+        expect(result.subCategories).to.be.empty;
+        expect(result.models).to.be.empty;
+      });
+
+      it("caches class type", async () => {
+        const classMock = {
+          is: (baseClass: string, baseSchema: string) => {
+            return baseClass === "Element" && baseSchema === "BisCore";
+          },
+        };
+
+        schemaMock.getClass.onCall(0).returns(Promise.resolve(classMock as unknown as ECClass));
+        schemaMock.getClass.onCall(1).throws();
```

Comment:
I think it would make test cleaner to assert `getClass` was called 1 time instead of mocking second call as error.